cmd/sqlflow/main.go

```diff
@@ -14,65 +14,343 @@
+	// TODO(yancey1989): finding a normative way to deal with that.
+	replacer := strings.NewReplacer(`\n`, "\n", `\t`, "\t", `\r`, "\r")
+	line = replacer.Replace(line)
+
+	for i = 0; i < len(line); i++ {
+		if isEscape {
+			isEscape = false
+			continue
+		}
+		switch line[i] {
+		case '\\':
+			if *inQuotedString {
+				isEscape = true
+			}
+		case '"', '\'':
+			if *inQuotedString {
+				if *isSingleQuoted == (line[i] == '\'') {
+					*inQuotedString = false // We found the end of a quoted string
+				}
+			} else { // The start of a quoted string
+				*inQuotedString = true
+				*isSingleQuoted = (line[i] == '\'')
+			}
+		case ';':
+			if !*inQuotedString { // We found a statement
+				if i-start != 1 { // Ignore empty statement that has only a ';'
+					(*statements)[len(*statements)-1] += line[start : i+1]
+				}
+				for i+1 < len(line) && isSpace(line[i+1]) {
+					i++ // Ignore leading whitespaces of the next statement
+				}
+				start = i + 1
+				if start == len(line) {
+					return true // All done, the last character in the line is the end of a statement
+				}
+				*statements = append(*statements, "") // Prepare for searching the next statement
+
+			}
+		case '-':
+			if !*inQuotedString {
+				if i+1 < len(line) && line[i+1] == '-' {
+					if i+2 == len(line) || isSpace(line[i+2]) { // We found a line comment
+						// Note: `--` comment doesn't interfere with quoted-string and `;`
+						(*statements)[len(*statements)-1] += strings.TrimSpace(line[start:i])
+						if len(*statements) == 1 && (*statements)[0] == "" {
+							*statements = []string{}
+							return true // The whole line is an empty statement that has only a `-- comment`,
+						}
+						return false
+					}
+				}
+			}
+		}
+	}
+	(*statements)[len(*statements)-1] += line[start:]
+	return false
+}
+
+// readStmt reads a SQL statement from the scanner.  A statement could have
+// multiple lines and ends at a semicolon at the end of the last line.
+func readStmt(scn *bufio.Scanner) ([]string, error) {
+	stmt := []string{}
+	var inQuotedString, isSingleQuoted bool
+	for scn.Scan() {
+		if addLineToStmt(scn.Text(), &inQuotedString, &isSingleQuoted, &stmt) {
+			return stmt, nil
+		}
+	}
+	// If the the file doesn't ends with ';', we consider the remaining content as a statement
+	if scn.Err() == nil {
+		return stmt, io.EOF
+	}
+	return stmt, scn.Err()
+}
+
+func flagPassed(name ...string) bool {
+	found := false
+	for _, n := range name {
+		flag.Visit(func(f *flag.Flag) {
+			if f.Name == n {
+				found = true
+			}
+		})
+	}
+	return found
+}
+
+func createRPCConn(serverAddr string) (*grpc.ClientConn, error) {
+	caCrt := os.Getenv("SQLFLOW_CA_CRT")
+	if caCrt != "" {
+		creds, _ := credentials.NewClientTLSFromFile(caCrt, "")
+		return grpc.Dial(serverAddr, grpc.WithTransportCredentials(creds))
+	}
+	return grpc.Dial(serverAddr, grpc.WithInsecure())
+}
+
+func sqlRequest(program string, ds string) *pb.Request {
```

Comment:
How to fill other fields of the Session struct?

Reply:
Currently, other fields are get from environment variable. You can export them or specify them in `.sqlflow_env`.